### DISCUSSION TITLE
Using Env variables inside package.json during next build command

### DISCUSSION THREAD

**Discussion by ahetawal-p** [vercel/next.js#39839] [Help] [answered]
We need to support different internal envs and not just the standard dev,prod,preview env which come from nextjs.
An example is different firebase config for local, dev, sandbox, prod.
We tried setting them up in package.json as something like this:

```
scripts: {
 "build:sandbox": "MY_ENV=sandbox && next build",
 "build:prod": "MY_ENV=production && next build",
}
```
And in one of config files we try to use like:
```
const sandbox = {
... // firebase app sandbox configs
}

const production = {
....// firebase app prod configs
}

console.info(`<<<< See Me >>> ${process.env.MY_ENV}`) // always says undefined
switch (process.env.MY_ENV) {
  case 'production':
    config = production
    break
  case 'sandbox':
    config = sandbox
    break
}

export default config
```
During the `next build:sandbox or next build:prod` phase we use `process.env.MY_ENV` is always coming in as undefined.
What is the best way to have custom configurations without being tied to nextjs standard envs ?

**Comment by icyJoseph** [comment, marked as answer]
Hi.

Maybe I am missing something here, but using `&&` breaks the command in two... and the variable won't be present, unless you set it globally, for the next command.

Try this instead, without `&&`:

```json
"build:sandbox": "MY_ENV=sandbox next build",
```

**Reply by ahetawal-p** [reply]
Ahh yes thank you, somehow I was just not seeing the extra `&&` ..

**Reply by gregg-cbs** [reply]
on windows i am getting "'NODE_ENV' is not recognized as an internal or external command,
operable program or batch file."

**Reply by icyJoseph** [reply]
On Windows you need to use `cross-env` https://www.npmjs.com/package/cross-env, or rather, learn how to pass env vars to windows, but `cross-env` makes that easier.
